### PR TITLE
refactor/adjust css variable to default pattern

### LINKED ISSUE #733: [REFACTOR] Adjust CSS variable to 'Default values' pattern
Acceptance Criteria
- walid components styles in storybook
- use 'default values' pattern for CSS variables [inspiration](https://css-tricks.com/patterns-for-practical-css-custom-properties-use/#article-header-id-2)
``` css
.sf-button {
  margin: var(--button-margin);
  padding: var(--button-padding, 1.5rem 2.5rem);
}
```
- update docs for 'default values' pattern
- update auto generation dosc scripts
- #445 limit number of CSS vars to certain properties

### PR DESCRIPTION
**Pull request by pspaczek**
# Related issue

Closes #733
# Scope of work

We decide to use CSS custom properties (variables) default patter in Storefront UI. In this PR we rewrite all components styles to this pattern and update **Contributing Guide/Coding guidelines/CSS Custom Properties** and **Customization/Intro to CSS Custom Properties**
# Screenshots of visual changes

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/shared/styles/components/molecules/SfAddToCart.scss [hunk 1]
```diff
@@ -1,34 +1,27 @@
 @import '../../variables';
-:root {
-  --add-to-cart__select-quantity-min-width: 4.625rem;
-  --add-to-cart__select-quantity-max-width: 5.625rem;
-}
+
 .sf-add-to-cart {
   display: flex;
   &__button {
     flex: 1;
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfAddToCart.scss]:
```suggestion
    flex: 1;
   order: var(--add-to-cart-button-order);
```
@martaradziszewska

File: packages/shared/styles/components/molecules/SfAddToCart.scss [hunk 2]
```diff
@@ -1,34 +1,27 @@
...
-    &__select-quantity {
-      margin-left: 0;
-      input {
-        font-size: var(font-size-extra-big);
-        background-color: var(--c-light);
+    input {
+      border-color: var(--add-to-cart-select-quantity-input-border-color, var(--c-light));
+      @include for-desktop {
+        font-size: var(--add-to-cart-select-quantity-input-f-size, var(--font-size-extra-big));
+        --input-background: var(--c-light);
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfAddToCart.scss]: @martaradziszewska first override variables and after that you add new property

File: packages/shared/styles/components/molecules/SfAddToCart.scss [hunk 3]
```diff
@@ -1,34 +1,27 @@
...
     }
-    &__select-quantity {
-      margin-left: 0;
-      input {
-        font-size: var(font-size-extra-big);
-        background-color: var(--c-light);
+    input {
+      border-color: var(--add-to-cart-select-quantity-input-border-color, var(--c-light));
+      @include for-desktop {
+        font-size: var(--add-to-cart-select-quantity-input-f-size, var(--font-size-extra-big));
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfAddToCart.scss]: @martaradziszewska use always font() when you want need even one font property example rules 19 in our docs
```
--input-font-size: var(--font-size-extra-big)
```
and for border
```
input{
--input-border: 1px solid var(--c-light)
}
```

File: packages/shared/styles/components/molecules/SfRadio.scss [hunk 4]
```diff
@@ -1,77 +1,85 @@
...
+
 .sf-radio {
-  $this: &;
-  display: flex;
-  font-family: var(--radio-font-family);
-  font-size: var(--radio-font-size);
-  font-weight: var(--radio-font-weight);
-  line-height: var(--radio-line-height);
-  transition: var(--radio-transition);
+  display: var(--radio-display, flex);
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfRadio.scss]: @aniamusial why you add this variable?

File: packages/shared/styles/components/molecules/SfRadio.scss [hunk 5]
```diff
@@ -1,77 +1,85 @@
...
   &.is-active {
-    background-color: var(--radio--is-active-background);
+    background: var(--radio-background, var(--c-light));
   }
-  &--transparent {
-    &.is-active {
-      background-color: transparent;
+
+  &__label {
+    display: var(--radio-label-display, flex);
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfRadio.scss]:
```suggestion
    display: flex;
```

File: packages/shared/styles/components/molecules/SfRadio.scss [hunk 6]
```diff
@@ -1,77 +1,85 @@
...
+    font-size: var(--radio-label-font-size, var(--font-size-extra-big));
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfRadio.scss]: @aniamusial use font(--radio-label-font, ...) here "19. use always font() when you want need even one font property"

File: packages/shared/styles/components/molecules/SfRadio.scss [hunk 7]
```diff
@@ -1,77 +1,85 @@
...
+    margin: var(--radio-description-margin, var(--spacer-medium) 0 0 0);
+  }
+
+  &--transparent {
+    &.is-active {
+      --radio-background: transparent;
+    }
+  }
+
+  @include for-desktop {
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfRadio.scss]: @aniamusial move @include for-desktop {} to top before l:20 :)

File: packages/shared/styles/components/molecules/SfBanner.scss [hunk 8]
```diff
@@ -1,97 +1,58 @@
...
-  padding: var(--banner-padding-y) var(--banner-padding-x);
-  color: var(--banner-color);
+  min-height: var(--banner-min-height, 9rem);
+  padding: var(--banner-padding, var(--spacer-big));
+  color: var(--banner-color, var(--c-text));
   background-repeat: no-repeat;
   background-position: bottom left;
   background-size: cover;
-  background-color: var(--background-color);
+  background-color: var(--background);
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfBanner.scss]:
```suggestion
  background-color: var(--banner-background);
```
@martaradziszewska

File: packages/shared/styles/components/molecules/SfBanner.scss [hunk 9]
```diff
@@ -1,97 +1,58 @@
...
+    font: var(--banner-subtitle-font, font(--banner-subtitle, 300, var(--font-size-regular), 1.16, var(--body-font-family-primary)));
+    color: var(--banner-subtitle-color, var(--c-dark-variant));
   }
   &__title {
-    font-size: var(--banner__title-font-size);
-    font-weight: var(--banner__title-font-weight);
+    font: var(--banner-title-font, font(--banner-title, 300, var(--font-size-extra-big), 1.6, var(--body-font-family-secondary)));
   }
   &__description {
     display: none;
```
Comment (pspaczek) [inline packages/shared/styles/components/molecules/SfBanner.scss]:
```suggestion
    display: none;
```
@martaradziszewska maybe we can use variable and move ale for-desktop to BLOCK scope?

Comment (pspaczek) [reply packages/shared/styles/components/molecules/SfBanner.scss]: and MODIFIER scope for modifiers :) like on example, this way is ok 👍 
```
 &--left {
    @include for-desktop {
      --banner-padding: 4rem calc(50% + 5rem) 4rem 4rem;
    }
  }
```

File: packages/shared/styles/components/molecules/SfReview.scss [hunk 10]
```diff
@@ -1,77 +1,81 @@
...
+    margin: var(--review-info-margin, 0 0 var(--spacer-medium) 0);
   }
+
   &__date {
-    text-transform: var(--review__date-text-transform);
-    font-weight: var(--review__date-font-weight);
-    font-size: var(--review__date-font-size);
-    color: var(--review__date-color);
+    text-transform: var(--review-date-text-transform, capitalize);
+    font-weight: var(
```
[the remark this answers is not in the document]
Comment (aniamusial) [reply packages/shared/styles/components/molecules/SfReview.scss]: indeed, I missed it :)

File: packages/vue/docs/contributing/coding-guidelines.md [hunk 11]
```diff
@@ -73,18 +73,157 @@ For example: `.gallery__item--selected`
...
 :::
 
+### CSS Custom Properties
 
-### CSS structure
+#### Rules and tips to create and use CSS Custom Properties / CSS Variables. 
+
+1. **DO NOT** use SASS VARIABLES.
+
+2. **Create variable** with default value like this:
```
Comment (pspaczek) [inline packages/vue/docs/contributing/coding-guidelines.md]: @martaradziszewska, you can create empty variable without default too:
```css
.sf-button {
    border-radius: var(--button-border-radius);
}

File: packages/vue/docs/contributing/coding-guidelines.md [hunk 12]
```diff
@@ -73,18 +73,157 @@ For example: `.gallery__item--selected`
...
+Create variables for all customizable properties like 
+`padding`, `margin`, `order`, `z-index`, `color`, `border`, `border-radius`, `background-color`. 
+
+Think of properties you would like to change it as a user and create for it a variable.
+
+
+Create size variables `var(--component-size, 25px)` in case when you have the same width and height value 
+
+```css
+.sf-button{
```
Comment (pspaczek) [inline packages/vue/docs/contributing/coding-guidelines.md]:
```suggestion
.sf-component{
```
@martaradziszewska or use SfArrow where you use square shape :)

File: packages/vue/docs/contributing/coding-guidelines.md [hunk 13]
```diff
@@ -73,18 +73,157 @@ For example: `.gallery__item--selected`
...
+:::tip
+SCSS functions and mixins you cand find here `shared/styles/_helpers.scss`
+
+All global variables are here `shared/styles/_variables.scss`
+:::
+
+4. **CSS Variables Naming convention**
+
+- variable name for BLOCK **var(--input-width);**
+- variable name for BLOCK elements **var(--input-label-padding);**
```
Comment (pspaczek) [inline packages/vue/docs/contributing/coding-guidelines.md]:
```suggestion
- variable name for ELEMENT **var(--input-label-padding);**
```
@martaradziszewska

File: packages/vue/docs/contributing/coding-guidelines.md [hunk 14]
```diff
@@ -73,18 +73,157 @@ For example: `.gallery__item--selected`
...
+      1.6,
+      var(--body-font-family-secondary)
+    )
+  );
+``` 
+
+- If you want to customize:
+```css
+.sf-banner {
+    &--slim {
```
Comment (pspaczek) [inline packages/vue/docs/contributing/coding-guidelines.md]: @martaradziszewska please add declaration for &__title{...} to make it more transparent 👍

File: packages/vue/docs/contributing/documentation-guidelines.md [hunk 15]
```diff
@@ -105,20 +105,21 @@ Multiline descriptions are merged into a single line, joined by dots (if necessa
 Use single line comments (with `//`) above variables for variables title. _Example:_
 
 ```css
-:root {
+  .sf-circle-icon {
     // Primary Button
```
Comment (pspaczek) [inline packages/vue/docs/contributing/documentation-guidelines.md]: @aniamusial @martaradziszewska do we use comments in our components?

**Pull request merged**